packages/core/content-releases/admin/src/components/CMReleasesInjectionZone.tsx

```diff
@@ -0,0 +1,280 @@
+import { Formik, Form } from 'formik';
+import { useIntl } from 'react-intl';
+import { useParams } from 'react-router-dom';
+import * as yup from 'yup';
+
+import { CreateReleaseAction } from '../../../shared/contracts/release-actions';
+import { PERMISSIONS } from '../constants';
+import { useCreateReleaseActionMutation, useGetReleasesQuery } from '../services/release';
+
+import { ReleaseActionOptions } from './ReleaseActionOptions';
+
+const RELEASE_ACTION_FORM_SCHEMA = yup.object().shape({
+  type: yup.string().oneOf(['publish', 'unpublish']).required(),
+  releaseId: yup.string().required(),
+});
+
+interface FormValues {
+  type: CreateReleaseAction.Request['body']['type'];
+  releaseId: CreateReleaseAction.Request['params']['releaseId'];
+}
+
+const INITIAL_VALUES = {
+  type: 'publish',
+  releaseId: '',
+} satisfies FormValues;
+
+interface AddActionToReleaseModalProps {
+  handleClose: () => void;
+}
+
+const AddActionToReleaseModal = ({ handleClose }: AddActionToReleaseModalProps) => {
+  const { formatMessage } = useIntl();
+  const toggleNotification = useNotification();
+  const { formatAPIError } = useAPIErrorHandler();
+  const params = useParams<{ id?: string }>();
+  const {
+    allLayoutData: { contentType },
+  } = useCMEditViewDataManager();
+  // Get all 'pending' releases
+  const { data } = useGetReleasesQuery({
+    filters: {
+      $and: [
+        {
+          releasedAt: {
+            $null: true,
+          },
+        },
+      ],
+    },
+  });
+  const releases = data?.data;
+  const [createReleaseAction, { isLoading }] = useCreateReleaseActionMutation();
+
+  const handleSubmit = async (values: FormValues) => {
+    /**
+     * contentType uid and entry id are not provided by the form but required to create a Release Action.
+     * Optimistically we expect them to always be provided via params and CMEditViewDataManager.
+     * In the event they are not, we should throw an error.
+     */
+    if (!contentType?.uid || !params.id) {
+      toggleNotification({
+        type: 'warning',
+        message: formatMessage({
+          id: 'content-releases.content-manager.notification.entry-error',
+          defaultMessage: 'Failed to get entry',
+        }),
+      });
+
+      return;
+    }
+
+    const releaseActionEntry = {
+      contentType: contentType.uid,
+      id: params.id,
+    };
+    const response = await createReleaseAction({
+      body: { type: values.type, entry: releaseActionEntry },
+      params: { releaseId: values.releaseId },
+    });
+
+    // When 'data' is in the response
+    if ('data' in response) {
+      // Handle success
+      toggleNotification({
+        type: 'success',
+        message: formatMessage({
+          id: 'content-releases.content-manager-edit-view.add-to-release.notification.success',
+          defaultMessage: 'Entry added to release',
+        }),
+      });
+    }
+    // When 'error' is in the response
+    if ('error' in response) {
```

Comment:
Another heads up, it looks like it would be possible here to have both. I guess it shouldn't actually happen in practice, but maybe you can return early above to be extra safe

Reply:
ok I had understood the response would either return error or data but I think you might be right, updated.